core/src/HIP/Kokkos_HIP_Parallel_Team.hpp

```diff
@@ -498,8 +498,16 @@ class ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties...>,
  public:
   __device__ inline void operator()() const {
     // Iterate this block through the league
-    int64_t threadid = 0;
-    if (m_scratch_size[1] > 0) {
+    int64_t threadid            = 0;
+    const int scratch_size_l1   = m_scratch_size[1];
+    const int shmem_begin       = m_shmem_begin;
+    const int shmem_size        = m_shmem_size;
+    void* scratch_level_1_ptr   = nullptr;
+    const size_type league_size = m_league_size;
+    int const int_league_size   = static_cast<int>(league_size);
+    void* __restrict__ const shmem_ptr =
+        ::Kokkos::Experimental::kokkos_impl_hip_shared_memory<void>();
```

Comment:
are you sure this makes any difference? Isn't the restrict getting lost again since we only use it to construct the team member? Same for moving int_league_size? there is only a single use of that thing why does it help to move this forward?

Reply:
Generally speaking: I can see how making local variables out of this stuff may change compiler optimization. But do we actually need to define this stuff up here instead of closer to the usage?

Reply:
Yeah, I was not too careful about tracking what in here specifically helped.  I _suspect_ it was mostly to do with moving the site of the division into the L1 scratch handling block.  Let me go re-test that and use that to inform what we should do RE: @dalg24's comment